ctapipe/core/container.py

```diff
@@ -190,18 +190,27 @@ def validate(self, value):
                     f"{errorstr} Should have dimensionality {self.ndim}"
                 )
             if value.dtype != self.dtype:
-                raise FieldValidationError(
-                    f"{errorstr} Has dtype "
-                    f"{value.dtype}, should have dtype"
-                    f" {self.dtype}"
-                )
+                if np.can_cast(value, self.dtype, casting="safe"):
+                    value = value.astype(self.dtype)
+                else:
+                    raise FieldValidationError(
+                        f"{errorstr} Has dtype "
+                        f"{value.dtype}, should have dtype"
+                        f" {self.dtype}"
+                    )
         else:
             # not a numpy array
             if self.dtype is not None:
-                if not isinstance(value, self.dtype.type):
-                    raise FieldValidationError(
-                        f"{errorstr} Should have numpy dtype {self.dtype}"
-                    )
+                # Workaround for string input
+                if np.issubdtype(type(value), np.number) and (
```

Comment:
I think this workaround is (at least at the moment) not needed, as we don't support writing string arrays and python str are handled via type, not dtype.

Reply:
Some form of workaround is still needed for the case of the Field requiring a non-string and the user submitting a string.

Reply:
See proposal above. Just reraise a `ValidationError("Validation failed") from err`